lib/global_config.js

```diff
@@ -114,9 +115,12 @@ exports.getOcspResponseCacheMaxAge = function () {
  *
  * @returns {string}
  */
-exports.mkdirCacheDir = function () {
-  let cacheRootDir = process.env.SF_OCSP_RESPONSE_CACHE_DIR;
-  if (!Util.exists(cacheRootDir)) {
+
+exports.mkdirCacheDir = function (envPath)
```

Comment:
using this logic with tokens creates a security vulnerability, since we don't want to store tokens in /tmp dir that can be accessed also by other users. Our logic should be that the directory can be only either `homedir()`or the one specified by the user (`envPath`), otherwise the feature should be disabled.

Reply:
The reason why I stored the token in this location is because this location is the place where we are saving the ocsp response cache file. Is the reason why we need to save it in homedir() or envPath because the 'securestorage' is more secure than ocsp_response_cache? If you confirm, I will redesign this.

Reply:
Also it would be great to create unit test that checks what is the actual cache directory

Reply:
I think token leak introduces much greater risk, but we should consider to exclude shared directories from  ocsp response cache too.

Reply:
I redesigned it. Instead of using mkdirCachedir in the global config, I make the default credential manager use home dir or env path. Because I need to check that either homedir() or the one specified is valid, I redesigned the json credential manager as a class and I implemented that it checks that token cache directory is whether valid or not. If it is not valid, all the methods (write, read, remove) just return null. Also, I added the unit testing for the token directory.